mail_editor/process.py

```diff
@@ -0,0 +1,75 @@
+import hashlib
+
+from lxml import etree
+
+
+def load_image(url):
+    import requests
+    # TODO check domains?
+    # TODO check data urls
+    r = requests.get(url)
+    content_type = r.headers["Content-Type"]
+    if not content_type.startswith("image/"):
+        raise Exception("not an image file")
+    return r.content, content_type
+
+
+def make_url_absolute(url, base_url=""):
+    """
+    base_url: https://domain
+    """
+    if not url:
+        if base_url:
+            return base_url
+        else:
+            return "/"
+    if "://" in url:
+        return url
+    if url[0] != "/":
+        url = f"/{url}"
+    return base_url + url
+
+
+def process_html(html, base_url="", extract_attachments=True, fix_links=True):
+    parser = etree.HTMLParser()
+    root = etree.fromstring(html, parser)
+
+    image_attachments = dict()
+    url_cid_cache = dict()
+
+    if extract_attachments:
```

Comment:
I'd recommend using mail-cleaner as dependency for this instead: https://github.com/maykinmedia/mail-cleaner/blob/main/mail_cleaner/mail.py#L92

It's your code from Open Forms in a separate package :)

Reply:
It is not quite the same; the linked code does data-urls, while this does http url's.

The intent is render standard Django templates with static and url tags and everything.

Anyway, I did isolate it a little bit so we can lift it to mail-cleaner later.